wandb/sdk/interface/artifacts.py

```diff
@@ -637,6 +637,66 @@ def wait(self) -> "Artifact":
         """
         raise NotImplementedError
 
+    def __getitem__(self, name: str) -> Optional[WBValue]:
+        """
+        Gets the WBValue object located at the artifact relative `name`.
+
+        NOTE: This will raise an error unless the artifact has been fetched using
+        `use_artifact` or the API.
+
+        Arguments:
+            name: (str) The artifact relative name to get
+
+        Raises:
+            Exception: if problem
+
+        Examples:
+            Basic usage
+            ```
+            artifact = wandb.Artifact('my_table', 'dataset')
+            table = wandb.Table(columns=["a", "b", "c"], data=[[i, i*2, 2**i]])
+            artifact["my_table"] = table
+
+            wandb.log_artifact(artifact)
+            ```
+
+            Retrieving an object:
+            ```
+            artifact = wandb.use_artifact('my_table:latest')
+            table = artifact["my_table"]
+            ```
+        """
+        return self.get(name)
```

Comment:
I would make this raise `NotImplementedError` and force an implementation in both `public.py` and `wandb_artifacts.py` as this is intended as an interface and I'm worried this makes it more of an abstract class.

I made a mistake with the others in this file (like `ArtifactManifest`) -- I really think they should implement 0 logic.

Reply:
i see - fair enough - not a problem. I'll update this